app/models.py

```diff
@@ -374,35 +380,50 @@ class SampleStageFile(db.Model):
     def sample_stage_id(self):
         return self.sample_stage.obfuscated_id
 
+    @hybrid_property
+    def status(self):
+        return FileStatus(self._status)
+
+    @status.setter
+    def status(self, s):
+        self._status = s.value
+
+    @status.expression
+    def status(cls):
+        return cls._status
+
     # create a sample stage file object
-    def __init__(self, sample_stage, status=FileStatus.incomplete):
-        kwds = {}
-        # set the sample stage
-        kwds['sample_stage'] = sample_stage
-        # construct the filename based on project, sample, and method
-        # TODO: There is no file extension...
-        method = sample_stage.method
-        sample = sample_stage.sample
+    def __init__(self, relative_upload_name, sample_stage, status=FileStatus.incomplete):
+
+        self.sample_stage = sample_stage
+
+        # construct the file path based on project, sample, and method
+        method  = sample_stage.method
+        sample  = sample_stage.sample
         project = sample.project
         relpath, counter = create_upload_filename(
             app.config['STORE_PATH'],
-            '{project_id:05d}'.format(project_id=project.id),
-            '{sample_id:05d}'.format(sample_id=sample.id),
-            '{method_id:05d}'.format(method_id=method.id))
-        kwds['relative_file_path'] = relpath
-        kwds['file_repr'] = \
+            'project-{project_id:05d}'.format(project_id=project.id),
+            'sample-{sample_id:05d}'.format(sample_id=sample.id),
+            'method-{method_id:05d}'.format(method_id=method.id),
+            os.path.basename(relative_upload_name))
+
+        self.relative_source_path = relative_upload_name
+        self.relative_target_path = relpath
+
+        # ???: Why are we storing this?
```

Comment:
No, it's not rhetorical.  I'm not convinced that it makes sense to store the representation, since string representations are usually more for human consumption than for machines.   Now that we're storing the actual paths independently, I can't think of a reason why we'd want the representation in the DB, too.